File: Data Structures & Algorithms/is-subsequence/submission-1.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        

        int n=s.length();
        int m=t.length();
        // int i=0,j=0;  
        // while(i<n&&j<m){
          
        //   if(s.charAt(i)==t.charAt(j)) i++;

        //   j++;
        // System.out.println(j+" "+i);
        // }
        // return i==n;
    
       return helper(s,t,n,m,0,0);
    }

    private boolean helper(String s,String t,int n,int m,int i,int j){
            
            if(i==n) return true;

            if(j==m) return false;

            if(s.charAt(i)==t.charAt(j)){
                return helper(s,t,n,m,i+1,j+1);
            }
            else
               return helper(s,t,n,m,i,j+1);



    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:

    void helper(TreeNode* root,int maxi,int &cnt){
        
        if(!root) return ;

        if(root->val>=maxi){
            maxi=root->val;
            cnt++;
        }

        helper(root->left,maxi,cnt);

        

        helper(root->right,maxi,cnt);


      




    }

    int goodNodes(TreeNode* root) {
         if (!root) return 0;
         int cnt=0;
         helper(root,root->val,cnt);
        
        

       return cnt;
    }
};
